app/models/reservation_scopes.rb

```diff
@@ -78,6 +78,11 @@ def self.included(base) # rubocop:disable MethodLength, AbcSize
       scope :overlaps_with_date, lambda { |date|
         where('start_date <= ? and due_date >= ?', date, date)
       }
+      scope :consecutive_with, lambda { |res|
```

Comment:
oh, yeah, I wasn't thinking about other types of overlap at all, I'll fix that.

I'll look into Query Objects in the meantime before the meeting!